test/utils/deepObject.test.js

```diff
@@ -0,0 +1,384 @@
+const { expect } = require('chai');
+const { toPath } = require('lodash');
+
+const logger = require('../../src/logger');
+const { deepObject } = require('../../src/utils');
+
+describe('[utils] deepObject', () => {
+	it('Should export all avaible actions', () => {
+		expect(deepObject.get).is.not.an('undefined');
+		expect(deepObject.pathsToArray).is.not.an('undefined');
+		expect(deepObject.getSimple).is.not.an('undefined');
+	});
+
+	describe('deepObject > pathToArray', () => {
+		const { pathsToArray, getSimple } = deepObject;
+
+		it('should work for empty input', () => {
+			const result = pathsToArray();
+			expect(result).to.be.an('array');
+			expect(result).have.lengthOf(0);
+		});
+
+		it('should work for empty string input', () => {
+			const result = pathsToArray('');
+			expect(result).to.be.an('array');
+			expect(result).have.lengthOf(0);
+		});
+
+		it('should work for one string', () => {
+			const result = pathsToArray('x.y.z');
+			expect(result).to.be.an('array');
+			expect(result).have.lengthOf(3);
+			expect(result[0]).to.equal('x');
+			expect(result[1]).to.equal('y');
+			expect(result[2]).to.equal('z');
+		});
+
+		it('should work for one string without dot noation', () => {
+			const result = pathsToArray('x');
+			expect(result).to.be.an('array');
+			expect(result).have.lengthOf(1);
+			expect(result[0]).to.equal('x');
+		});
+
+		it('should work for strings with a lot of dot noations', () => {
+			const result = pathsToArray('ele0.ele1.ele2.ele3.ele4.ele5.ele6.ele7.ele8.ele9.ele10');
+			expect(result).to.be.an('array');
+			expect(result).have.lengthOf(11);
+			expect(result[0]).to.equal('ele0');
+			expect(result[10]).to.equal('ele10');
+		});
+		// the implementation is slower, becouse it can resolve mutch more combinations
+		// and loadash is optimize for re-runs with momory cache
+		it('[profiling] pathsToArray - '
+			+ 'should not significantly slower for string notations like lodash.toPath.', () => {
+			const testPath = 'ele0.ele1.ele2.ele3.ele4.ele5.ele6.ele7.ele8.ele9.ele10';
+			const iterations = 100000;
+			const slowerFactor = 10;
+
+			const s3 = Date.now();
+			for (let i = 0; i <= iterations; i += 1) {
+				testPath.split('.');
+			}
+			const baseSplitOperation = Date.now() - s3;
+
+			const s2 = Date.now();
+			for (let i = 0; i <= iterations; i += 1) {
+				toPath(testPath);
+			}
+			const loadashToPathDelta = Date.now() - s2;
+
+			const s1 = Date.now();
+			for (let i = 0; i <= iterations; i += 1) {
+				pathsToArray(testPath);
+			}
+			const pathToArrayDelta = Date.now() - s1;
+	
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Trailing spaces not allowed. (no-trailing-spaces)](https://app.codacy.com/app/schul-cloud/schulcloud-server/pullRequest?prid=4646529)